routing/route.cpp

```diff
@@ -354,6 +359,42 @@ void Route::Update()
   m_currentTime = 0.0;
 }
 
+void Route::AppendTraffic(Route const & route)
+{
+  CHECK(route.IsValid(), ());
+
+  if (GetTraffic().empty() && route.GetTraffic().empty())
+    return;
+
+  if (!IsValid())
+  {
+    m_traffic.assign(route.GetTraffic().cbegin(), route.GetTraffic().cend());
+    return;
+  }
+
+  // Note. At this point the last item of |m_poly| should be removed.
+  // So the size of |m_traffic| should be equal to size of |m_poly.
+  if (GetTraffic().empty())
+  {
+    CHECK_GREATER_OR_EQUAL(m_poly.GetPolyline().GetSize(), 1, ());
```

Comment:
Such cases are processed above in this method. 

```c++
void Route::AppendTraffic(Route const & route)
{
  CHECK(route.IsValid(), ());

  if (GetTraffic().empty() && route.GetTraffic().empty())
    return;

  if (!IsValid())
  {
    m_traffic.assign(route.GetTraffic().cbegin(), route.GetTraffic().cend());
    return;
  }

  // Note. At this point the last item of |m_poly| should be removed.
  // So the size of |m_traffic| should be equal to size of |m_poly|.
  if (GetTraffic().empty())
  {
    CHECK_GREATER_OR_EQUAL(m_poly.GetPolyline().GetSize(), 1, ());
    m_traffic.resize(m_poly.GetPolyline().GetSize(), SpeedGroup::Unknown);
```

If IsValid() returns false there's a special branch for it. And ```bool IsValid() const { return (m_poly.GetPolyline().GetSize() > 1); }```.